src/electionguard/elgamal.cpp

```diff
@@ -577,14 +578,19 @@ namespace electionguard
 
         unique_ptr<ElementModP> g_to_r = nullptr;
         unique_ptr<ElementModP> publicKey_to_r = nullptr;
-        // check if the are precompute values rather than doing the exponentiations here
-        unique_ptr<Triple> triple = PrecomputeBufferContext::getTriple();
-        if (triple != nullptr) {
-            g_to_r = triple->get_g_to_exp();
-            publicKey_to_r = triple->get_pubkey_to_exp();
-        } else {
-            g_to_r = g_pow_p(nonce);
 
+        if (shouldUsePrecomputedValues) {
```

Comment:
only try to use precompute if explicitly asked, and fall back gracefully to realtime if none exist. This way, we have more consistent behavior and we can at least explicitly force the caller to make the decision to use the buffers and take on the risk of side effects.